contracts/src/systems/combat/contracts.cairo

```diff
@@ -668,6 +686,10 @@ mod combat_systems {
             let mut battle: Battle = get!(world, battle_id, Battle);
             battle.update_state();
 
+            if battle.has_ended() {
```

Comment:
replace with `assert!` would be clearer no?

Reply:
yeah done